File: backend/models/restaurant.js
const mongoose = require('mongoose')

const model = mongoose.Schema({
    name: {
        type: String,
        required: true
    },
    address: {
        type: String,
        required: false
    },
    city: {
        type: String,
        required: false
    },
    postcode: {
        type: String,
        required: false
    },
    latitude: {
        type: Number,
        required: false
    },
    longitude: {
        type: Number,
        required: false
    },
    brand: {
        type: mongoose.ObjectId,
        required: true
    },
    createdBy: {
        type: mongoose.ObjectId,
        required: true
    },
    creationDate: {
        type: Date,
        required: true
    }
});

module.exports = new mongoose.model("Restaurant", model)

File: webapp/lib/models/restaurant.dart
import 'package:latlong2/latlong.dart';

class Restaurant {
  late String id;
  late String name;
  late String address;
  late String city;
  late String postcode;
  late LatLng coordinates;

  Restaurant(this.id, this.name, this.address, this.city, this.postcode,
      this.coordinates);

  Restaurant.fromJson(Map<String, dynamic> json)
      : id = json['_id'],
        name = json['name'],
        address = json['address'],
        city = json['city'],
        postcode = json['postcode'],
        coordinates = LatLng(json['latitude'], json['longitude']);
}
